utils/file/file.go

```diff
@@ -261,7 +261,7 @@ func OverwriteFile(file string, data []byte) {
 	if err != nil {
 		log.Errorln("Error creating file")
 	}
```

Comment:
The change to close the file at all times looks fine. But here it seems that if we fail to create the file we still seem to continue on and try to write to the non-existent file. We should return early if we fail to create the file, no?